src/rgw/rgw_rados.cc

```diff
@@ -6205,7 +6205,7 @@ int RGWRados::swift_versioning_restore(RGWObjectCtx& obj_ctx,
    *
    * TODO: delegate this check to un upper layer and compare with ACLs. */
   if (bucket_info.owner != archive_binfo.owner) {
-    return -EPERM;
+    return 0;
```

Comment:
It look that Swift silently ignores the restore part if user lacks privileges. If we want to do the same, returning `0` is not enough. We also need to assign `false` to `restored`. Otherwise we can face some random behavior dependent on what was previously stored in the underlying memory.